File: 800/2037A/main.java
import java.util.*;

public class main 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();
        
        while (t-- > 0) 
        {
            int n = sc.nextInt();
            
            Map<Integer, Integer> mp = new HashMap<>();
            int cnt = 0;
            
            int[] arr = new int[n];
            for (int i = 0; i < n; i++)
                arr[i] = sc.nextInt();
            
            for (int i = 0; i < n; i++) 
            {
                mp.put(arr[i], mp.getOrDefault(arr[i], 0) + 1);
                
                if (mp.get(arr[i]) == 2) 
                {
                    cnt++;
                    mp.put(arr[i], 0); // reset after forming a pair
                }
            }
            
            System.out.println(cnt);
        }
    }
}

File: 800/2037A/main.cpp
#include <bits/stdc++.h>
using namespace std;
 
int main() 
{
    int t;
    cin>>t;
    
    while(t--)
    {
        int n;
        cin>>n;
        
        unordered_map<int, int> mp;
        int cnt = 0;
        
        int arr[n];
        for(int i=0;i<n;i++)
            cin>>arr[i];
        
        for(int i=0;i<n;i++)
        {
            if(mp[arr[i]])
                mp[arr[i]] = mp[arr[i]] + 1;
            else
                mp[arr[i]] = 1;
            
            if(mp[arr[i]] == 2)
            {
                cnt++;
                mp[arr[i]] = 0;
            }
        }
        
        cout<<cnt<<endl;
    }
 
    return 0;
}